### ISSUE TITLE
Infinite loop with `Layout/FirstHashElementIndentation`, `Layout/HashAlignment`, `Layout/ExtraSpacing`, `Layout/FirstArgumentIndentation`, `SpaceInsideParens` cops

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#14853]
## Steps to reproduce the problem

Run RuboCop with autocorrect on the following code:

```ruby
describe License, type: :model do
  let(:license) do
    License.new({
    multiple_occupation: multiple_occupation,
  issued_under_act_part_yes: issued_under_act_part_yes,
  issued_under_act_part_applied: issued_under_act_part_applied,
  issued_by: issued_by
  }.merge(
    form_date('issued_date', issued_date)
  ))
  end

  subject { license }

  context 'licence applied for' do
    let(:multiple_occupation)           { 'Applied' }
    let(:issued_under_act_part_yes)     { '' }
    let(:issued_by)                     { '' }
    let(:issued_date)                   { '' }

    context 'no part 2 or part 3 selected' do
      let(:issued_under_act_part_applied) { '' }
      it { should_not be_valid }
    end
  end
end
```

## Expected behavior

It should not enter an infinite correction loop.

## Actual behavior

> 1 error occurred:
Infinite loop detected in /home/ydakuka/Work/rubocop/file.rb and caused by Layout/FirstHashElementIndentation, Layout/HashAlignment -> Layout/FirstArgumentIndentation, Layout/HashAlignment -> Layout/ExtraSpacing, Layout/FirstArgumentIndentation, Layout/FirstHashElementIndentation, Layout/SpaceInsideParens -> Layout/ExtraSpacing, Layout/FirstHashElementIndentation, Layout/HashAlignment, Layout/SpaceInsideParens -> Layout/FirstArgumentIndentation, Layout/HashAlignment -> Layout/FirstArgumentIndentation, Layout/FirstHashElementIndentation

--------

## RuboCop version

```
$ bundle exec rubocop -V
1.84.1 (using Parser 3.3.10.0, rubocop-ast 1.49.0, analyzing as Ruby 2.7, running on ruby 3.4.8) [x86_64-linux]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.9.0
```

**Comment by koic** [comment]
This issue is a dup with #14858.
